packages/core/src/components/button/_button-group.scss

```diff
@@ -142,11 +142,20 @@ Styleguide button-group
   }
 
   &.#{$ns}-outlined {
+    > .#{$ns}-popover-target > .#{$ns}-button,
     > .#{$ns}-button {
       @include pt-button-outlined();
     }
 
+    &.#{$ns}-vertical {
+      > .#{$ns}-popover-target:not(:last-child) .#{$ns}-button,
+      > .#{$ns}-button:not(:last-child) {
+        border-bottom: none;
+      }
+    }
+
     &:not(.#{$ns}-vertical) {
+      > .#{$ns}-popover-target:not(:last-child) > .#{$ns}-button,
       > .#{$ns}-button:not(:last-child) {
         border-right: none;
       }
```

Comment:
Changes to this file include a subset of the same changes originally applied in #6966, but scoped only to the `outlined` ButtonGroup.